### PR TITLE
blake2s cpp and rust API

### PR DESCRIPTION
**Pull request by emirsoyturk**
## Changes

This PR updates the existing Blake2s implementation by integrating a C++ API and a Rust wrapper API.

### CODE REVIEW

File: icicle/src/hash/CMakeLists.txt [hunk 1]
```diff
@@ -1,5 +1,10 @@
 set(TARGET icicle_hash)
 
-add_library(${TARGET} STATIC keccak/extern.cu)
+set(SRC ${CMAKE_SOURCE_DIR}/src/hash)
+
+set(HASH_SOURCE ${SRC}/keccak/extern.cu)
```
Comment (yshekel) [inline icicle/src/hash/CMakeLists.txt:5]: minor but you could do both in same line rather then use append

File: wrappers/rust/icicle-hash/src/blake2s/tests.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        let height = 22;
+        let input_block_len = 136;
+        let leaves = vec![1u8; (1 << height) * input_block_len];
+        let mut digests = vec![0u64; merkle_tree_digests_len((height + 1) as u32, 2, 1)];
+
+        let leaves_slice = HostSlice::from_slice(&leaves);
+        let digests_slice = HostSlice::from_mut_slice(&mut digests);
+
+        build_blake2s_merkle_tree(leaves_slice, digests_slice, height, input_block_len, &config).unwrap();
+        println!("Root: {:?}", digests_slice[0]);
```
Comment (yshekel) [inline wrappers/rust/icicle-hash/src/blake2s/tests.rs:48]: note that both tests don't test correctness

**Review by yshekel — APPROVED** [review]

**Comment by jeremyfelder** [comment]
@emirsoyturk @yuvalingo It seems like the base branch is from V2. Is this PR meant to be against V3? I think this [PR](https://github.com/ingonyama-zk/icicle/pull/610)/branch is the latest for blake2s on V3

**Comment by emirsoyturk** [comment]
@jeremyfelder right. Maybe we can merge it into the `aviad/blake2s` branch, and then I can open another branch for V3. That way, we would have Blake2s for both V2 and V3.

**Pull request merged**